File: app/src/main/java/de/sneakpeek/data/MovieInfoProductionCompany.java
package de.sneakpeek.data;


import android.os.Parcel;
import android.os.Parcelable;

import com.squareup.moshi.Json;

public class MovieInfoProductionCompany implements Parcelable {

    public final static Parcelable.Creator<MovieInfoProductionCompany> CREATOR = new Creator<MovieInfoProductionCompany>() {


        @SuppressWarnings({
                "unchecked"
        })
        public MovieInfoProductionCompany createFromParcel(Parcel in) {
            MovieInfoProductionCompany instance = new MovieInfoProductionCompany();
            instance.name = ((String) in.readValue((String.class.getClassLoader())));
            instance.id = ((int) in.readValue((int.class.getClassLoader())));
            return instance;
        }

        public MovieInfoProductionCompany[] newArray(int size) {
            return (new MovieInfoProductionCompany[size]);
        }

    };
    @Json(name = "name") public String name;
    @Json(name = "id") public int id;

    public void writeToParcel(Parcel dest, int flags) {
        dest.writeValue(name);
        dest.writeValue(id);
    }

    public int describeContents() {
        return 0;
    }

}

File: app/src/main/java/de/sneakpeek/data/MovieInfoProductionCompany.kt
package de.sneakpeek.data


import android.os.Parcel
import android.os.Parcelable
import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName

data class MovieInfoProductionCompany(
        @Expose @SerializedName("name") var name: String?,
        @Expose @SerializedName("id") var id: Int = 0) : Parcelable {

    companion object {
        @JvmField val CREATOR: Parcelable.Creator<MovieInfoProductionCompany> = object : Parcelable.Creator<MovieInfoProductionCompany> {
            override fun createFromParcel(source: Parcel): MovieInfoProductionCompany = MovieInfoProductionCompany(source)
            override fun newArray(size: Int): Array<MovieInfoProductionCompany?> = arrayOfNulls(size)
        }
    }

    constructor(source: Parcel) : this(
            source.readString(),
            source.readInt()
    )

    override fun describeContents() = 0

    override fun writeToParcel(dest: Parcel, flags: Int) {
        dest.writeString(name)
        dest.writeInt(id)
    }
}